### PR TITLE
Cleanup Github Actions and Makefile

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

Improve the terminology used in some of our files to make it easier for developers

# Context

## Push to Solo-Apis Github Action
There has been ongoing confusion about how to manually trigger the github action that publishes Gloo changes to solo-apis, off of a branch. This is due to poor naming on my part when I initially built the job. 

The previous names worked in the context of a release, but are confusing in the context of a manual trigger. I have updated the names to be more representative.

I would like to have @sheidkamp and @ianmacclancy review. Seth raised this initially and I don't believe Ian has run the job before so he will be a perfect test for whether this is clear.

### Testing
I trigged the job from this branch, using the following inputs:
Screen Shot 2023-04-21 at 8 36 17 AM

The logs are here:https://github.com/solo-io/gloo/actions/runs/4764928981

The automation worked, and PR was created: https://github.com/solo-io/solo-apis/pull/803

## Makefile Improvements
This is a copy/paste of the work that merged in Gloo Enterprise with: https://github.com/solo-io/solo-projects/pull/4785/

I also removed some targets from the `make help` output which are not run by developers. If we want to add them back, we can. I want to make sure that make help provides guidance for commands that devs run regularly.

I would like to have @jbohanon review this as he made the original changes.

### Testing
```
➜ make help
L21   help                                Output the self-documenting make targets
L198  install-go-tools                    Download and install Go dependencies
L240  test                                Run all tests, or only run the test package at {TEST_PKG} if it is specified
L251  run-tests                           Run all non E2E tests, or only run the test package at {TEST_PKG} if it is specified
L259  run-ci-regression-tests             Run the Kubernetes E2E Tests in the {KUBE2E_TESTS} package
L308  generated-code                      Execute Gloo Edge codegen
L780  kind-build-and-load-%               Use to build specified image and load it into kind
L788  kind-reload-%                       Use to build specified image, load it into kind, and restart its deployment
L814  kind-list-images                    List solo-io images in the kind cluster named {CLUSTER_NAME}
L819  kind-prune-images                   Remove images in the kind cluster named {CLUSTER_NAME}
L853  scan-version                        Scan all Gloo images with the tag matching {VERSION} env variable
L27   glooctl                             Builds the command line tool
L41   build-cli-local                     Build the CLI according to your local GOOS and GOARCH
```

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sam-heilbron** [comment]
/skip-changelog This is a minor CI/docs change, without a tracking issue

**Comment by sam-heilbron** [comment]
/kick https://github.com/solo-io/gloo/issues/8003#issuecomment-1517787778

File: .github/workflows/README.md [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
+### 1. On a Release
+Whenever Gloo Edge is released, we want to automatically update the published APIs in our read-only mirror. This action will be triggered, and after a release completes a developer needs to manually approve the PR in solo-apis.
+
+### 2. On a Manual Trigger
+We may want to test API changes in other projects, without actually merging the API changes into Gloo Edge. To do this, we push the API changes to a branch in Gloo, and then use this workflow to publish the changes to solo-apis.
+
+The arguments are:
+- `Use Workflow From`: The branch which contains the workflow code you want to execute. Often times, the default branch will work.
+- `Source Branch`: The branch which contains the API changes you want to mirror
+- `Target Branch`: The Gloo LTS branch which you want to merge your changes into
```
[the remark this answers is not in the document]
Comment (sam-heilbron) [reply .github/workflows/README.md:22]: Yeah so the way the job works, the input is the gloo lts branch, `v1.13.x`. The job then performs the "magic" to build the solo-apis branch from that. The hope is that this means the developer just puts in inputs that it cares about (what is the branch im copying from, and what is the branch im trying to copy to)

File: Makefile [hunk 2]
```diff
@@ -613,7 +614,7 @@ ifeq ($(RELEASE), "false")
 endif
 
 .PHONY: generate-helm-files
-generate-helm-files: $(OUTPUT_DIR)/.helm-prepared ## Creates Chart.yaml and values.yaml. See install/helm/README.md for more info.
+generate-helm-files: $(OUTPUT_DIR)/.helm-prepared
```
Comment (jbohanon) [inline Makefile:617]: I agree this target probably doesn't need to be documented for humans. This makes me wonder though if we _should_ have self-doc on the `build-test-chart` target, which I use extensively in both OSS and EE

Comment (sam-heilbron) [reply Makefile:617]: ha great minds! i thought about doing this, went to the code, started to think about the message and then got distrcted on a call. I'd like to do this and I'll push aan update

File: .github/workflows/README.md [hunk 3]
```diff
@@ -1,15 +1,27 @@
...
+
+**NOTE: After the PR opens in solo-apis, we want to avoid the chance that it merges. Please put a 'work in progress' label on the PR to prevent it from merging.**
```
[the remark this answers is not in the document]
Comment (sam-heilbron) [reply .github/workflows/README.md:24]: Good call! Ill push something up, and then if there are further improvements we can make, perhaps we hop on a call and go through the proposal together just as a faster way of incorporating the ideas

[the remark this answers is not in the document]
Comment (sam-heilbron) [reply .github/workflows/README.md:24]: Pushed updates with examples, let me know what you think

File: .github/workflows/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (sam-heilbron) [reply .github/workflows/README.md:22]: I'm happy to try to clarify this further if there is still confusion. Maybe a call would make the most sense

File: Makefile (hunk 2 above)
Comment (sam-heilbron) [reply Makefile:617]: pushed an update, let me know if there's more i can add

**Review by jbohanon — APPROVED** [review]

**Review by ianmacclancy — APPROVED** [review]

File: .github/workflows/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sheidkamp) [reply .github/workflows/README.md:36]: lgtm!

**Review by sheidkamp — APPROVED** [review]

**Review by jbohanon — APPROVED** [review]

**Pull request merged**